kustomize/internal/commands/edit/add/secret.go

```diff
@@ -139,4 +139,7 @@ func mergeFlagsIntoGeneratorArgs(args *types.GeneratorArgs, flags flagsAndArgs)
 		args.EnvSources = append(
 			args.EnvSources, flags.EnvFileSource)
 	}
+	if flags.Behavior != "" {
```

Comment:
We could, but I'm not too familiar with secret generattion so I didnt want to take any chances. I think it is the same change in secret.go as in configmap.go but maybe @mortent can weigh in.

Reply:
If we want to hold off on a similar improvement for _Secrets_, could we remove this change to file _secret.go_? It was hard to tell whether you needed to make this change in order to keep the code working as it had been.

Reply:
No no the reason I made the change in secret.go is because the mergeFlagsIntoGeneratorArgs is called from configmap.go. Its like a common utility function.

Reply:
Oh! It's hard to see that without a local copy of the code base, given the narrow lens that GitHub's diff-oriented code review interface presents. I just saw the file name _secret.go_ and thought, "Ah, we're now taking care of _Secrets_ too."